components/region_cache_memory_engine/src/engine.rs

```diff
@@ -458,10 +630,25 @@ impl Peekable for RegionCacheSnapshot {
         cf: &str,
         key: &[u8],
     ) -> Result<Option<Self::DbVector>> {
-        Ok(self.region_memory_engine.data[cf_to_id(cf)]
-            .get(key)
-            .cloned()
-            .map(|v| RegionCacheDbVector(v)))
+        let seq = self.sequence_number;
+        let mut iter = self.region_memory_engine.data[cf_to_id(cf)].iter();
+        let seek_key = encode_seek_key(key, self.sequence_number, VALUE_TYPE_FOR_SEEK);
+
+        iter.seek(&seek_key);
+        if !iter.valid() {
+            return Ok(None);
+        }
+        let InternalKey {
```

Comment:
nit: it may be possible to simplify match arm and avoid early return using a pattern guard (if user_key == &key). not clear if it actually makes this more reasonable, so i'll leave the decision to you. 
Example:
```rust
Ok(match x { 
    InternalKey { 
         user_key, v_type: ValueType::Value, .. 
    } if user_key == &key => 
        Some(RegionCacheDbVector(iter.value().clone())),
    _  => None,
})
```

Reply:
Good suggestions and modified.